packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -33,16 +40,30 @@ export class Wallet {
   }
 
   private _messageService: MessageServiceInterface;
-
+  private _destroyed = false;
   private constructor(
     messageServiceFactory: MessageServiceFactory,
     private _engine: Engine,
     private _retryOptions: RetryOptions
   ) {
+    super();
     const handler: MessageHandler = async message => {
-      const {outbox} = await this._engine.pushMessage(message.data);
-
-      await this.messageService.send(getMessages(outbox));
+      try {
+        const {outbox, newObjectives} = await this._engine.pushMessage(message.data);
+        // Receiving messages from other participants may have resulted in new proposed objectives
+        for (const o of newObjectives) {
+          this.emit('ObjectiveProposed', o);
+        }
+        await this.messageService.send(getMessages(outbox));
+      } catch (err) {
```

Comment:
This is the change that I believe resolves the flickering test. By swallowing errors after the `wallet` is destroyed we prevent abort errors from bubbling up.

I think this is a relatively safe change since it should only swallow errors after the `wallet` has been destroyed.

Reply:
Is there a use case for destroying a wallet in production?

Reply:
This change has now been removed since 18f62e8 resolves the issue of `pushMessage` executing during `destroy`.